test/suites/basic.sh

```diff
@@ -1173,3 +1173,149 @@ EOF
 
   services_validator
 }
+
+test_remove_cluster_member() {
+  unset_interactive_vars
+
+  microcloud_internal_net_addr="$(ip_config_to_netaddr lxdbr0)"
+
+  # Set the default config for interactive setup.
+  export LOOKUP_IFACE="enp5s0"
+  export LIMIT_SUBNET="yes"
+  export EXPECT_PEERS=2
+  export SETUP_ZFS="yes"
+  export ZFS_FILTER="lxd_disk1"
+  export ZFS_WIPE="yes"
+  export SETUP_CEPH="yes"
+  export SETUP_CEPHFS="yes"
+  export CEPH_FILTER="lxd_disk2"
+  export CEPH_WIPE="yes"
+  export CEPH_CLUSTER_NETWORK="${microcloud_internal_net_addr}"
+  export CEPH_ENCRYPT="no"
+  export SETUP_OVN="yes"
+  export OVN_FILTER="enp6s0"
+  export IPV4_SUBNET="10.1.123.1/24"
+  export IPV4_START="10.1.123.100"
+  export IPV4_END="10.1.123.254"
+  export DNS_ADDRESSES="10.1.123.1,8.8.8.8"
+  export IPV6_SUBNET="fd42:1:1234:1234::1/64"
+
+  reset_systems 3 3 3
+  echo "Fail to remove member from MicroCeph and LXD until OSDs are removed"
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+
+  # Wait for roles to refresh from the next heartbeat.
+  for i in $(seq 1 40) ; do
+    if lxc exec micro01 --env "TEST_CONSOLE=0" -- microcloud cluster list | grep -q PENDING ; then
+      sleep 1
+    else
+      break
+    fi
+  done
+
+  ! lxc exec micro01 --env "TEST_CONSOLE=0" -- microcloud remove micro02 || true
+
+  for s in "microcloud" "microovn" ; do
+    lxc exec micro01 --env "TEST_CONSOLE=0" -- ${s} cluster list | grep -q "micro01"
+    ! lxc exec micro01 --env "TEST_CONSOLE=0" -- ${s} cluster list | grep -q "micro02" || true
+    lxc exec micro01 --env "TEST_CONSOLE=0" -- ${s} cluster list | grep -q "micro03"
+  done
+
+  for s in "lxc" "microceph" ; do
+    lxc exec micro01 --env "TEST_CONSOLE=0" -- ${s} cluster list | grep -q "micro01"
+    lxc exec micro01 --env "TEST_CONSOLE=0" -- ${s} cluster list | grep -q "micro02"
+    lxc exec micro01 --env "TEST_CONSOLE=0" -- ${s} cluster list | grep -q "micro03"
+  done
```

Comment:
What is happening in between the two loops that the second check for `micro02` doesn't require `|| true`?

Reply:
I missed the loop vars... nvm :)